src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <hyro/utils/SpinnerDefault.h>
+#include <hyro/factory/CommandFactory.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/SignalGenerator.h>
+#include <hyro/msgs/Signal.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+using namespace std::chrono_literals;
+
+namespace hyro
+{
+std::shared_ptr<HyroLogger> SignalGeneratorComponent::s_logger = HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+std::shared_ptr<ChannelOutput<Signal>> m_signal_output;
+hyro::DynamicPropertyAccess dynamic_property_access_signal("/signal"_uri);
+SignalGenerator sg;
+
+Result
+SignalGeneratorComponent::init (const ComponentConfiguration & configuration)
+{
+  registerDynamicProperty<double>("amplitude",
+                                  &SignalGeneratorComponent::setAmplitude,
+                                  &SignalGeneratorComponent::getAmplitude,
+                                  this
+                                  );
+
+  registerDynamicProperty<double>("frequency",
+                                  &SignalGeneratorComponent::setFrequency,
+                                  &SignalGeneratorComponent::getFrequency,
+                                  this
+                                  );
+
+  registerDynamicProperty<bool>("cosine",
+                                &SignalGeneratorComponent::setCosine,
+                                &SignalGeneratorComponent::getCosine,
+                                this
+                                );
+
+  std::shared_ptr<ChannelOutput<std::vector<int>>>
+  m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, configuration);
```

Comment:
You don't need to use `this` all the time. I usually just use `this` when it is required.